recipes-omnect/initrdscripts/omnect-os-initramfs/factory-reset

```diff
@@ -216,70 +243,78 @@ factory_reset_run() {
         return 1
     fi
 
-    factory_reset_mode=$(get_bootloader_env_var "${FACTORY_RESET_VAR}")
-    if [ -z "${factory_reset_mode}" ]; then
-        return  # no factory reset triggered; normal boot
+    local mode=$(echo "${factory_reset}" | jq -e -r .mode) || { handle_status "1" "-" "mode_not_set" ""; return 0; }
+    # don't error if preserve is not set
+    local preserve=$(echo "${factory_reset}" | jq -r '.preserve')
+    if [[ "null" != "${preserve}" ]]; then
+        preserve=($(echo "${factory_reset}" | jq -e -r '.preserve | .[]')) || { handle_status "3" "-" "could_not_parse_preserve_array" ""; return 0; }
+    else
+        preserve=()
     fi
 
-    factory_reset_restore_list=$(get_bootloader_env_var "${FACTORY_RESET_RESTORE_LIST_VAR}")
-
-    # we already mounted /dev/omnect/boot when we're using grub. so we unmount so we don't disturb the restore mount/umount handling
-    if [ -f /usr/bin/grub-editenv ]; then
-        sync
-        run_cmd umount ${ROOTFS_DIR}/boot
+    msg "Running factory-reset [mode=${mode}, preserve=${preserve[@]}]..."
+    factory_reset_mount
+
+    # create preserve_list
+    declare -a preserve_list
+    preserve_list+=("/etc/omnect/factory-reset.d/")
+    if [[ " ${preserve[@]} " =~ " applications " ]]; then
+        preserve=( ${preserve[@]/applications} )
+        for i in $(find ${ROOTFS_DIR}/etc/omnect/factory-reset.d/ -iname *.json); do
+            preserve_list+=( $(cat ${i} | jq -r -e '.paths[]') )
```

Comment:
no need for `cat`, could be replaced with simple stdin redirection from file

Reply:
true ... do you see a special benefit in doing it that way?

Reply:
avoidance of extra process invocation could be an argument here.
but to me use of file redirection always appeared the Linux way and an unneeded pipe doesn't make things clearer, at least here the simple `< ${i}` looks better imho.
it might feel different in a longer pipe construct, in the end this is a question of preference.
my 0,02 €.

Reply:
here i could even change it to (and will so)
```
preserve_list+=( $(jq -r -e '.paths[]' ${i}) )
```
without an pipe like `|`or `<`. 
from the result it imho doesn't make a difference though.
"extra process invocation" may be an argument, is this an established rule though?

Reply:
Will be handled in a future PR.